### DISCUSSION TITLE
"External cannot be loaded due to macOS quarantine" at install time

### DISCUSSION THREAD

**Discussion by ideoforms** [playbenny/benny#462] [General]
When I first run Benny on macOS (Ventura on Apple M1), I get the below warning, once for each external:

<img width="288" alt="image" src="https://github.com/user-attachments/assets/7d24ff9f-c667-445c-8afa-0a694b17dda8">

It is definitely a [known issue](https://cycling74.com/forums/%22external-cannot-be-load-due-to-macos-quarantine%22) with Max, seemingly due to [Gatekeeper](https://cycling74.com/articles/using-unsigned-max-externals-on-mac-os-10-15-catalina) security policies. It's ultimately benign, as you can click "Apply to all" and "OK" and it will remove the quarantine attribute, but it would be nice if there were some way to address this. It seems like the [idea](https://superuser.com/questions/28384/what-should-i-do-about-com-apple-quarantine) of the quarantine attribute is to prevent malware. 

I wondered whether you might be able to remove the attribute, zip the file up, and then distribute the un-quarantined version — but of course, this would be a massive security hole, and macOS re-adds it when you download the new zip.

So, I guess the options are:
 - leave it as-is and flag it up as a warning in the first-run docs; or
 - add a "Run Benny" script, which, when run, removes quarantine attributes and then opens the Max patch; or a "Remove quarantine" script

The latter would be pretty easy, but adds some clutter to the filesystem and means that macOS users have to follow a different process by running this script. So it's possibly simplest just to add this to the docs for now, and maybe later address this with code signing if you decide to create a proper signed app release of Benny. What do you think?

**Comment by ideoforms** [comment]
interestingly, this happens when you download the .zip from GitHub, but doesn't seem to happen if you do a `git clone`. I guess these quarantine flags are added in the Chrome/web download flow.

**Comment by jamesholdenmusic** [comment]
Thank you for collecting all this information!
I think you're right - I'll add it to the docs, mention that updating it via git circumvents it, and when the time comes to package up a more accessible version of benny we can deal with it properly.
